### PR TITLE
change: use SimpleTemplate

### PR DESCRIPTION
**Pull request by eddieantonio**
This is an experiment to make dealing with `<template>` and `<slot>` elements easier. It's a proposal, but if there are any objections, feel free to close this PR.

This introduces the `SimpleTemplate` class, which is a tiny wrapper around the `<template>` DOM interface. Say you have the following template in the HTML:

```html
<template id="template:recording-link">
   <li class="recording__link">
     <button>
       <slot name="speaker">Unknown speaker</slot>, <slot name="dialect">unknown dialect</slot>
     </button>
   </li>
</template>
```

If you want to fill it out with `recordingData`, instantiate a `SimpleTemplate` from its `id`:

```js
let recordingLink = SimpleTemplate.fromId('template:recording-link')
```

Then fill in the text inside the named `<slot>`s:

```js
recordingLink.slot.speaker = recordingData['speaker_name']
recordingLink.slot.dialect = 'Maskwacîs'
```

Finally, insert the filled-in template into the page:

```js
let container = document.querySelector('.recording__list')
container.appendChild(recordingLink.element)
```

@kobexamoh, if the code doesn't make sense, or you don't like it, feel free to close this!

### CODE REVIEW

**Comment by kobexamoh** [comment]
Oh wow oh _wow_ this looks really incredible. I'll look with fresh eyes in the morning too (I see that there's conflicting files right now) but I imagine this speeds up the work immensely!

File: src/simple-template.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ *       greeting.slot.recipient = 'cruel world'
+ *
+ * Now, insert the template into the DOM:
+ *
+ *       document.body.appendChild(greeting.element)
+ *
+ * This will result with the following inserted into the page:
+ *
+ * <body>
+ *    <h1> <slot name="salutation">Goodbye</slot>, <slot name="recipient">cruel world</slot>!</h1>
```
Comment (kobexamoh) [inline src/simple-template.js:39]: [tfw](https://youtu.be/SUtf9Ajlno4?t=226)

**Comment by eddieantonio** [comment]
> All of this looks good to me: the new template file is _really_ interesting: I didn't know you could do object-oriented stuff on the frontend like that – I'm very intrigued by this.
> 
> Since there's a merge conflict I'm guessing this can't be merged right away: do you have to tweak anything to make it alright for the main branch?

I went ahead and resolved it! Yeah, I knew I was going to make a merge conflict with myself on #459 since I was modifying the exact same lines, but luckily, the fixes were easy enough!

**Comment by kobexamoh** [comment]
All looks good then – merging!

**Pull request merged**